File: Program_B.py
# Program takes in a list of numbers from user and displays the largest number in the list


# Function takes in a list of numbers and returns the largest number in the list
def findMax(numList):
    i = 0
    j = len(numList)
    max = 0

    while not (i > j):
        if numList[i] > max:
            max = numList[i]
        if numList[j] > max:
            max = numList[j]
        
        i += 1
        j -= 1

    return max

# main Function
if __name__ == '__main__':
    inputList = []

    while True:
        i = input("Enter a number to store. Enter 0 to exit input step: ")

        if i == '0':
            break
        else:
            inputList.append(int(i))

    print("Largest number entered: " + str(findMax(inputList)))

File: Program_B.cpp
/*
Program takes in a dynamically growing list of numbers from user and displays the biggest number found in the list
*/

#include <iostream>
#include <string>

// function prototypes
void growArray(int*, int);
int findMax(int*, int);

// main function
int main() {
    int size = 0;
    int* numList = new int[size];

    std::string input;
    while (true) {
        input = "";
        std::cout << "Enter a number to store. Enter 0 to exit number input: ";
        std::cin >> input;

        if (input == "0") break;
        else {
            growArray(numList, size);

            numList[size] = std::stoi(input);
            size++;
        }
    }

    std::cout << "Largest number in list: " << findMax(numList, size) << std::endl;
    return 0;
}

// Function takes in address for array and allocates new memory for it to grow dynamically
void growArray(int* arr, int size) {
     
    int* newArr = new int[size + 1];

    if (size > 0) {
        for (int i = 0; i < size; i++) {
            newArr[i] = arr[i];
        }
    }
    
    arr = newArr; 
}

// Function finds the largest number in the list
int findMax(int* arr, int size) {
    int max = 0;
    int i = 0;
    int j = size - 1;

    while (!(i > j)) {
        if (arr[i] > max) max = arr[i];
        if (arr[j] > max) max = arr[j];

        i++;
        j--;
    }

    return max;
}